### PR TITLE
Refactor result output

### LINKED ISSUE #92: Result file output could use better naming conventions
Laundry list of issues:
- [x] remove failure 
- [x] add observed vs expected intent
- [x] remove test and transcript from expected_entities
- [x] add test and transcript to top level
- [x] drop test failures
- [x] drop character error rate
- [x] parse schema into additional entities with "schema" 
- [x] split intent and entity accuracy

### PR DESCRIPTION
**Pull request by tshastry**
Fixes #92 

New result file output:
```
{
  "0": {
    "expected_intent": "inquiry",
    "observed_intent": "inquiry",
    "expected_entities": {
      "fruit_description": "green tomatoes",
      "fruit_type": "tomatoes"
    },
    "observed_entities": {
      "fruit_description": "green tomatoes",
      "fruit_type": "tomatoes"
    },
    "test_name": "Inquiry",
    "transcript": "what is your price on green tomatoes"
  },
  ...
  "total_tests": 4,
  "test_time_sec": 0.06,
  "expected_intents": [
    "inquiry",
    ...
  ],
  "observed_intents": [
    "inquiry",
    ...
  ],
  "total_entity_errors": 0,
  "total_entities": 10,
  "correct_tests": 4,
  "test_file": "testing/interpreter_test/success.txt",
  "entity_accuracy": 100.0,
  "test_accuracy": 100.0
}
```

Old output:
```
{
  "0": {
    "test_name": "Inquiry",
    "transcript": "what is your price on green tomatoes",
    "expected_intent": "inquiry",
    "observed_intent": "inquiry",
    "failure": 0,
    "test_failures": [],
    "expected_entities": {
      "test": "Inquiry",
      "transcript": "what is your price on green tomatoes",
      "intent": "inquiry",
      "fruit_description": "green tomatoes",
      "fruit_type": "tomatoes"
    },
    "observed_entities": {
      "fruit_description": "green tomatoes",
      "fruit_type": "tomatoes"
    }
  },
  ...
  "total_tests": 4,
  "test_time_sec": 0.05,
  "expected_intents": [
    "inquiry",
    ...
  ],
  "observed_intents": [
    "inquiry",
    ...
  ],
  "total_entity_character_errors": 0,
  "total_character_errors": 0,
  "total_characters": 125,
  "correct_tests": 4,
  "test_file": "testing/interpreter_test/success.txt",
  "accuracy": 100.0
}
```

### CODE REVIEW

File: test_discovery.py [hunk 1]
```diff
@@ -62,10 +62,27 @@
...
+):
+    for k, v in json.loads(test_dict['schema']).items():
+        intent_results["expected_entities"]["schema_" + k] = v
+        intent_results["observed_entities"]["schema_" + k] = v
+
+    schema_failures = [
+        f[4] for f in entity_results['test_failures'] if f[2] == "schema"
+    ]
+
+    for k, v in json.loads(test_dict['schema']).items():
```
Comment (ageojo) [inline test_discovery.py]: same as above

File: test_discovery.py [hunk 2]
```diff
@@ -96,12 +113,33 @@ def test_one(test_dict, intent_whitelist, domain_whitelist):
...
 
     return y_true, y_pred, time_dif_ms, intent_results, entity_results
 
 
+def did_test_fail(entity_results, intent_results):
+    return (
+        1 if (
+          'expected_intent' in intent_results and \
+          intent_results['expected_intent'] != intent_results['observed_intent'] \
+        ) or ( \
```
Comment (ageojo) [inline test_discovery.py]: don't think the `\` are necessary as it is in parantheses

File: testing/evaluate_tests.py [hunk 3]
```diff
@@ -224,14 +215,14 @@ def test_single_case(
             errors, error_value = test_schema(
                 full_response, json.loads(expected_entity_value), test_name
             )
+            total_entities += len(json.loads(expected_entity_value).keys())
```
Comment (ageojo) [inline testing/evaluate_tests.py]:
```suggestion
            total_entities += len(json.loads(expected_entity_value))
```

File: test_discovery.py (hunk 2 above)
Comment (mgoldey) [reply test_discovery.py]: @ageojo this overrides `yapf` when `\` is present

Comment (tshastry) [reply test_discovery.py]: Yep, as Matthew said this is just to prevent a strange yapf I saw

File: test_discovery.py (hunk 1 above)
Comment (tshastry) [reply test_discovery.py]: In this case, we specifically need the key value pairs.

Comment (tshastry) [reply test_discovery.py]: I see what you mean now. This is a typo — it should be iterating over the schema failures

**Pull request merged**